### PR TITLE
lxd init: ask for images.auto_update_cached

### LINKED ISSUE #2167: Prompt to enable/disable image autoupdate
* Distribution: Ubuntu
 * Distribution version: 16.04
 * The output of "lxc info" or if that fails:
   * Kernel version: 4.4.0-28-generic
   * LXC version: 2.0.1-0ubuntu1~16.04.1 (driverversion: 2.0.1)
   * LXD version: 2.0.2-0ubuntu1~16.04.1 (serverversion: 2.0.2)
   * Storage backend in use:  ZFS

# Issue description

I'm using `lxd` on my laptop (Ubuntu desktop) on a daily basis for development purposes - I install all the needed tools (php, apache, nodejs, mongo, etc) in containers so I don't bloat my desktop environment and I can nuke anything at any time without worrying that there'll be anything left.

Currently I'm on a metered network (mobile hotspot) for a couple of days and I have a limited amount of traffic available (not even using video in conference calls). Today as I was working I noticed network activity and started to quickly close desktop applications to find the culprit. I closed everything and the network was still being actively used. Using `nethogs` and `iftop` I was able to spot `lxd` was happily eating lots of my precious traffic.

While I totally understand the benefit of having up to date images (especially from the security point of view) and the fact that this is a real edge case I would suggest to prompt to enable the image auto-update on `lxd init`. The main benefit would be that first the users will be aware that the container images do auto-update. I for one, had no idea about this but thankfully I found this information on  @stgraber 's blog (https://www.stgraber.org/2016/03/15/lxd-2-0-installing-and-configuring-lxd-212/). This might be useful to sysadmins that are setting up environments using provision tools (or setting up test environments in which usually the results are expected to be the same).

Secondly, I noticed a behavior which I'm not sure if it qualifies as a bug. The issue is that once an image download is initiated there's no way to stop it, i.e. it does not respect the `images.auto_update_cached` setting anymore (or maybe it does not respect it at all).

### Steps to reproduce

 1. Initiate the image auto update (can't tell how, is there a cron job so one could use `run-parts`?)
 2. `service lxd stop`
 3. `lxc config set images.auto_update_cached false`
 4. `service lxd start`
 5. Monitor network activity and notice that upon service start, the image download process started as well.

In my case, I had to delete both images so I can use my network again. Hopefully I will not need to spin-up any containers during these days.

Thank you for your patience and sorry for the long issue.

### PR DESCRIPTION
**Pull request by brauner**
On lxd init, ask the user whether he wants lxd to automatically refresh stale
images. Defaults to true.

Signed-off-by: Christian Brauner <christian.brauner@canonical.com>

Closes #2167.

### CODE REVIEW

File: lxd/main.go [hunk 1]
```diff
@@ -947,6 +953,18 @@ they otherwise would.
 		}
 	}
 
+	if imagesAutoUpdate {
+		_, err = c.SetServerConfig("images.auto_update_cached", "true")
```
Comment (tych0) [inline lxd/main.go]: I think you could shorten all this by using fmt.Sprintf("%s", imagesAutoUpdate) if you wanted.

Comment (stgraber) [inline lxd/main.go]: Nope, because "%s" of a boolean is rendered as "%!s(bool=VALUE)"

**Comment by stgraber** [comment]
So, changing auto_update_cached isn't actually the right thing to do here.

Changing that boolean controls whether further image imports will have the auto_update property set to true. It doesn't control whether existing images will be updated and more importantly, it's painful to the re-enable.

What this option should be doing is set images.auto_update_interval to 0 which simply disables the auto-update goroutine in a way that can then be easily re-enabled by changing it to whatever value you want.

File: lxd/main.go (hunk 1 above)
Comment (tych0) [inline lxd/main.go]: Ok, %v then.

Comment (stgraber) [inline lxd/main.go]: Oh right, that would work, except that as I commented, we shouldn't be setting images.auto_update_cached in the first place and the key we should set is an integer :)

**Comment by stgraber** [comment]
There is still one problem left with this branch. If I run "lxd init" twice, the first time I turn off image auto update and the second I don't, then it'll still be off.

So we need to be a bit more clever here, most likely:
 - Ask the question
 - If yes, set to 0
 - If no and, read the current value. If it's 0, then unset the key, if not, leave the current value alone.

**Comment by stgraber** [comment]
Looks like there is another issue. If used in auto mode (--auto), then the image update feature will always be turned off as the question won't be asked and the boolean defaults to false.

**Comment by brauner** [comment]
Sorry, will take care of this.

**Comment by stgraber** [comment]
Looks good to me. Running "lxd init --auto" will now effectively re-enable image auto-updates if it was previously disabled, but that's actually consistent with everything else that we do in "lxd init" so that's fine.

Will merge once Jenkins is happy.

Thanks!

**Comment by brauner** [comment]
Thanks for the patience. :)

**Pull request merged**